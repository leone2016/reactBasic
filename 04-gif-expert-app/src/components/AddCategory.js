import React, { useState } from 'react'
import PropTypes from 'prop-types';

 const AddCategory = (props) => {
    const { setcategories } = props;
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
        console.log('handleInputChange Called');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2 ){
            setcategories(cat=> [ inputValue, ...cat]);
            setInputValue('');
        }
       
    }
    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.prototype = {
    setcategories : PropTypes.func.isRequired
}


export default AddCategory;
