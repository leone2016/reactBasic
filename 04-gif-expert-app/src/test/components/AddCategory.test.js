import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import AddCategory from '../../components/AddCategory';


describe('UNIT TEST', () => {

    const setcategories = jest.fn();//()=>{};
    let wrapper = shallow(<AddCategory setcategories={setcategories} />);
    
    beforeEach( ()=>{
        jest.clearAllMocks(); // si se tiene algun mock o una  funcion, esto hara que se limpie
         wrapper = shallow(<AddCategory setcategories={setcategories} />);

    })
    test('should be well <AddCategory /> ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be  change the  inputText/> ', () => {
        const input = wrapper.find('input');
        const value =  'valor input';
        input.simulate('change', { target: {value} });
        expect( wrapper.find('p').text().trim()).toBe(value);
    });

    test('should NOT post the information with submit ', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setcategories).not.toHaveBeenCalled();
    });

    /**
     * tarea 
     * 1.- simular el inputChange
     * 2. simular el submit
     * 3. setcategories se debe de haber llamado
     * 4. limpiar el input text
     */
    test('should call setcategories and clean the input text ', () => {
        const value =  'valor input';
        wrapper.find('input').simulate('change',  { target: {value} });
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setcategories).toHaveBeenCalled();
        expect(setcategories).toHaveBeenCalledTimes(1);
        expect(setcategories).toHaveBeenCalledWith( expect.any(Function));
        expect( wrapper.find('input').prop('value')).toBe('');
    });
})