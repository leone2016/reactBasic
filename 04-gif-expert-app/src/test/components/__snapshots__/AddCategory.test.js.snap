// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UNIT TEST should be well <AddCategory />  1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
